skbeam/core/roi.py

```diff
@@ -691,3 +693,54 @@ def lines(end_points, shape):
         label += 1
         label_array[rr, cc] = label
     return label_array
+
+
+def auto_find_center_rings(avg_img, sigma=1, no_rings=4, min_samples=3,
+                           residual_threshold=1):
```

Comment:
it is better to add max_trials as an input to this function, so we can control from outside.